### PR TITLE
libc: provide implementation for open, read, close

### PR DESCRIPTION
**Pull request by oflebbe**
Implement simple wrappers for open, read and close for libc backed system.

Not too sure about data types to use.
At least it seems to work on MacOS/Darwin, though very slow.

### CODE REVIEW

File: src/syscall/syscall_libc.go [hunk 1]
```diff
@@ -26,15 +31,29 @@ func Write(fd int, p []byte) (n int, err error) {
...
 
 func Seek(fd int, offset int64, whence int) (off int64, err error) {
 	return 0, ENOSYS // TODO
 }
 
 func Open(path string, mode int, perm uint32) (fd int, err error) {
-	return 0, ENOSYS // TODO
+	buf, count := splitSlice([]byte(path))
+	cstr := libc_strndup(buf, count)
+	defer libc_free(cstr)
```
Comment (aykevl) [inline src/syscall/syscall_libc.go:49]: You can avoid a heap allocation here by avoiding the string to []byte conversion. You can create a new `splitString` function and pass the resulting pointer directly to `libc_strndup`.

File: src/syscall/syscall_libc.go [hunk 2]
```diff
@@ -76,10 +95,28 @@ func splitSlice(p []byte) (buf *byte, len uintptr) {
 	return slice.buf, slice.len
 }
 
+// int open(const char *path, int oflag, int perm)
+//export open
+func libc_open(path *byte, oflag int, perm uint32) int
```
Comment (aykevl) [inline src/syscall/syscall_libc.go:100]: `int` in C is practically always 32-bit (int32) on both 32-bit and 64-bit systems, while `int` in Go is 64-bit on 64-bit systems. Therefore, this signature is subtly wrong in an almost-compatible way. This is better:

```suggestion
func libc_open(path *byte, oflag int32, perm uint32) int32
```

**Review by aykevl — COMMENTED** [review]
The types look almost correct.

>  At least it seems to work on MacOS/Darwin, though very slow.

How did you test this?
It would be great to have some tests for this, possibly including Linux.

File: src/syscall/syscall_libc.go (hunk 2 above)
Comment (oflebbe) [reply src/syscall/syscall_libc.go:100]: Where is a suitable place in the file tree for a test ? 
Can we use go test framework ?

**Comment by deadprogram** [comment]
Is there any duplication with this PR and https://github.com/tinygo-org/tinygo/pull/1659 ?

**Comment by oflebbe** [comment]
Oh yes silly me. Didn't checked the open PR's first . Seems like  #1659 seem to contain my proposal  .

**Comment by oflebbe** [comment]
Withdrawing PR in favor of #1659

**Pull request closed without merging**